### PR TITLE
Move loop trade and verify balances of reserves into one function

### PR DESCRIPTION
**Pull request by manhlx3006**

### CODE REVIEW

File: contracts/sol6/KyberNetwork.sol [hunk 1]
```diff
@@ -579,71 +575,61 @@ contract KyberNetwork is WithdrawableNoModifiers, Utils5, IKyberNetwork, Reentra
...
         IERC20 src,
-        uint256 srcAmount,
         IERC20 dest,
-        uint256 conversionRate,
         uint256 srcDecimals,
         uint256 destDecimals
     ) internal
     {
+        // only need to verify src balance if src is not ETH
+        uint256 srcBalanceBefore = (src == ETH_TOKEN_ADDRESS) ? 0 : getBalance(src, address(this));
```
Comment (Anyhowclick) [inline contracts/sol6/KyberNetwork.sol:593]: can remove (src == ETH_TOKEN_ADDRESS)
`uint256 srcBalanceBefore = getBalance(src, address(this));`

Comment (manhlx3006) [reply contracts/sol6/KyberNetwork.sol:593]: if src token is ETH, we don't need to verify src token balance as we will pass this value to reserve's trade function

Comment (Anyhowclick) [reply contracts/sol6/KyberNetwork.sol:593]: Yes, so it should be the same as the line below right?
`uint256 srcBalanceBefore = (src == ETH_TOKEN_ADDRESS) ? 0 : getBalance(src, address(this));`
should be
`uint256 srcBalanceBefore = getBalance(src, address(this));`
like the line below?

Comment (manhlx3006) [reply contracts/sol6/KyberNetwork.sol:593]: if we don't need to verify src token balance, why do we need to get network's balance of src token?

Comment (Anyhowclick) [reply contracts/sol6/KyberNetwork.sol:593]: Ahhhhhhhh righhtttt, ok got it

**Comment by manhlx3006** [comment]
Bytecode size only increases 4 compare to current Katalyst branch

File: contracts/sol6/KyberNetwork.sol [hunk 2]
```diff
@@ -579,71 +581,61 @@ contract KyberNetwork is WithdrawableNoModifiers, Utils5, IKyberNetwork, Reentra
...
+            require(
+                reservesData.addresses[i].trade{value: callValue}(
+                    src,
+                    reservesData.srcAmounts[i],
+                    dest,
+                    address(this),
+                    reservesData.rates[i],
+                    true
+                ),
+                "trade failed"
```
Comment (ilanDoron) [inline contracts/sol6/KyberNetwork.sol:613]: maybe better
reserve trade failed

so its different from 
kyber trade failed.

File: test/sol6/kyberNetwork.js [hunk 3]
```diff
@@ -612,38 +612,30 @@ contract('KyberNetwork', function(accounts) {
...
-                    "reserve list is empty"
-                )
+                let tx = await tempNetwork.listReservesForToken(
+                    newToken.address, 0, 0, true, {from: operator}
+                );
+                // no event should be emitted
+                Helper.assertEqual(tx.receipt.logs.length, 0);
             });
 
             it("test list reserves empty as wrong indices", async() => {
```
Comment (ilanDoron) [inline test/sol6/kyberNetwork.js:631]: update title.

File: test/sol6/kyberNetwork.js [hunk 4]
```diff
@@ -612,38 +612,30 @@ contract('KyberNetwork', function(accounts) {
...
+                    newToken.address, 4, 5, true, {from: operator}
                 )
-                await expectRevert(
-                    tempNetwork.listReservesForToken(
-                        newToken.address, 4, 4, true, {from: operator}
-                    ),
-                    "reserve list is empty"
+                await tempNetwork.listReservesForToken(
+                    newToken.address, 4, 4, true, {from: operator}
                 )
```
Comment (ilanDoron) [inline test/sol6/kyberNetwork.js:639]: also here should check no event.
no?

File: contracts/sol6/KyberNetwork.sol (hunk 2 above)
Comment (manhlx3006) [reply contracts/sol6/KyberNetwork.sol]: updated

File: test/sol6/kyberNetwork.js (hunk 3 above)
Comment (manhlx3006) [reply test/sol6/kyberNetwork.js]: updated

File: test/sol6/kyberNetwork.js (hunk 4 above)
Comment (manhlx3006) [reply test/sol6/kyberNetwork.js:641]: updated

**Pull request merged**